tests/integration/distribution.go

```diff
@@ -690,197 +696,92 @@ func (s *CCVTestSuite) TestIBCTransferMiddleware() {
+		return sdk.NewDecCoinsFromCoins(bal...)
+	}
 
-	for _, tc := range testCases {
-		s.Run(tc.name, func() {
-			ctx, _ := s.providerCtx().CacheContext()
-			// fund consumer rewards pool
-			bankKeeper.SendCoinsFromAccountToModule(
-				ctx,
-				s.providerChain.SenderAccount.GetAddress(),
-				providertypes.ConsumerRewardsPool,
-				tc.rewardsPool,
-			)
+	totalRewards := sdk.Coins{sdk.NewCoin(sdk.DefaultBondDenom, math.NewInt(100))}
 
-			// update consumer rewars allocation
-			providerKeeper.SetConsumerRewardsAllocation(
-				ctx,
-				chainID,
-				providertypes.ConsumerRewardsAllocation{
-					Rewards: tc.rewardsAlloc,
-				},
-			)
+	// fund consumer rewards pool
+	bankKeeper.SendCoinsFromAccountToModule(
+		s.providerCtx(),
+		s.providerChain.SenderAccount.GetAddress(),
+		providertypes.ConsumerRewardsPool,
+		totalRewards,
+	)
 
-			// store pool balance
-			oldPool := bankKeeper.GetAllBalances(
-				ctx,
-				distributionKeeper.GetDistributionAccount(ctx).GetAddress(),
-			)
+	// Allocate rewards evenly between consumers
+	rewardsPerConsumer := totalRewards.QuoInt(math.NewInt(int64(len(s.consumerBundles))))
+	for chainID := range s.consumerBundles {
+		// update consumer allocation
+		providerKeeper.SetConsumerRewardsAllocation(
+			s.providerCtx(),
+			chainID,
+			providertypes.ConsumerRewardsAllocation{
+				Rewards: sdk.NewDecCoinsFromCoins(rewardsPerConsumer...),
+			},
+		)
+	}
 
-			coinsTransferred, _ := providerKeeper.GetConsumerRewardsAllocation(ctx, chainID).Rewards.TruncateDecimal()
-			// // transfer consumer rewards to distribution module
-			// coinsTransferred, err := providerKeeper.TransferConsumerRewardsToDistributionModule(
-			// 	ctx,
-			// 	chainID,
-			// )
-			// if tc.expErr {
-			// 	s.Require().Error(err)
-			// 	return
-			// }
-
-			// check remaining consumer rewards allocation
-			expCoinTransferred, expRemaining := tc.rewardsAlloc.TruncateDecimal()
-			s.Require().Equal(expCoinTransferred, coinsTransferred)
-
-			s.Require().Equal(
-				expRemaining,
-				providerKeeper.GetConsumerRewardsAllocation(ctx, chainID).Rewards,
-			)
+	// Iterate over the validators and
+	// store their current voting power and outstanding rewards
+	lastValOutRewards := map[string]sdk.DecCoins{}
+	totalValsRewards := sdk.DecCoins{}
+	for _, val := range s.providerChain.Vals.Validators {
+		valRewards, err := distributionKeeper.GetValidatorOutstandingRewards(s.providerCtx(), sdk.ValAddress(val.Address))
+		s.Require().NoError(err)
+		lastValOutRewards[sdk.ValAddress(val.Address).String()] = valRewards.Rewards
+		totalValsRewards = totalValsRewards.Add(valRewards.Rewards...)
+	}
 
-			// check updated consuemer rewards pool balance
-			newPool := bankKeeper.GetAllBalances(
-				ctx,
-				distributionKeeper.GetDistributionAccount(ctx).GetAddress(),
-			)
+	s.Require().True(totalValsRewards.IsZero())
 
-			s.Require().Equal(newPool.Sub(oldPool...), coinsTransferred)
-		})
+	// store community pool balance
```

Comment:
Doesn't `getDistrAcctBalFn` get the distribution module's balance? Is this the same as that of the community pool?

Reply:
Yes, in this setup the distribution module's account balance only holds the tokens that belong to the community pool. Note that the community pool is only a state of the distribution module, not a separate account.